src/net/sourceforge/kolmafia/KoLAdventure.java

```diff
@@ -356,11 +356,11 @@ public void overrideAdventuresUsed(int used) {
   // access to (some of) the features, if we didn't see you use the item which
   // granted daily access, we can usually visit a map area and take a look.
   //
-  // this.isValidAdventure will be false if the only way to visit an area is to
-  // use a (possibly expensive) day pass - which you might not even own
+  // This returns false if the only way to visit an area is to use a (possibly
+  // expensive) day pass - which you might not even own
 
   private boolean validate0() {
```

Comment:
Yes. They are now each called in exactly one place. I considered doing that this PR, but I liked having the comments describing the three stages of validation - each of which is used internally and also is exposed to scripts - close together rather than separated by large methods.

So it's a documentation issue.

(Another potential issue is that although the three methods are intended to be called in sequence and, with earlier steps short circuiting on failure, later steps assume earlier ones pass, scripts don't have to follow that. A script could call prepare_for_adventure without having called can_adventure, say. That's a bug in the script. I'm inclined to not try to protect against buggy scripts.)